### ISSUE TITLE
Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by kyrie111** [Tencent/Hunyuan3D-1#13]
### Env
refer to env_install.sh
[same ERROR both on H100(80G) and A6000(48G)]

### cmd
python3 main.py --image_prompt "demos/example_001.png" --save_folder ./outputs/test/ --max_faces_num 90000 --do_texture_mapping --do_render

### log
2024-11-06 17:11:15.701068: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
/root/miniconda3/envs/myconda/lib/python3.10/site-packages/diffusers/models/vq_model.py:20: FutureWarning: `VQEncoderOutput` is deprecated and will be removed in version 0.31. Importing `VQEncoderOutput` from `diffusers.models.vq_model` is deprecated and this will be removed in a future version. Please use `from diffusers.models.autoencoders.vq_model import VQEncoderOutput`, instead.
  deprecate("VQEncoderOutput", "0.31", deprecation_message)
/root/miniconda3/envs/myconda/lib/python3.10/site-packages/diffusers/models/vq_model.py:25: FutureWarning: `VQModel` is deprecated and will be removed in version 0.31. Importing `VQModel` from `diffusers.models.vq_model` is deprecated and this will be removed in a future version. Please use `from diffusers.models.autoencoders.vq_model import VQModel`, instead.
  deprecate("VQModel", "0.31", deprecation_message)
2024-11-06 17:11:20.066823729 [E:onnxruntime:Default, env.cc:254 ThreadMain] pthread_setaffinity_np failed for thread: 4025, index: 3, mask: {3, }, error code: 22 error msg: Invalid argument. Specify the number of threads explicitly so the affinity is not set.
2024-11-06 17:11:20.066826975 [E:onnxruntime:Default, env.cc:254 ThreadMain] pthread_setaffinity_np failed for thread: 4024, index: 2, mask: {2, }, error code: 22 error msg: Invalid argument. Specify the number of threads explicitly so the affinity is not set.
2024-11-06 17:11:20.066827568 [E:onnxruntime:Default, env.cc:254 ThreadMain] pthread_setaffinity_np failed for thread: 4023, index: 1, mask: {1, }, error code: 22 error msg: Invalid argument. Specify the number of threads explicitly so the affinity is not set.
2024-11-06 17:11:20.066830346 [E:onnxruntime:Default, env.cc:254 ThreadMain] pthread_setaffinity_np failed for thread: 4022, index: 0, mask: {0, }, error code: 22 error msg: Invalid argument. Specify the number of threads explicitly so the affinity is not set.
2024-11-06 17:11:20.066952981 [E:onnxruntime:Default, env.cc:254 ThreadMain] pthread_setaffinity_np failed for thread: 4027, index: 5, mask: {5, }, error code: 22 error msg: Invalid argument. Specify the number of threads explicitly so the affinity is not set.
2024-11-06 17:11:20.066903663 [E:onnxruntime:Default, env.cc:254 ThreadMain] pthread_setaffinity_np failed for thread: 4026, index: 4, mask: {4, }, error code: 22 error msg: Invalid argument. Specify the number of threads explicitly so the affinity is not set.
2024-11-06 17:11:20.067672883 [E:onnxruntime:Default, env.cc:254 ThreadMain] pthread_setaffinity_np failed for thread: 4028, index: 6, mask: {6, }, error code: 22 error msg: Invalid argument. Specify the number of threads explicitly so the affinity is not set.
2024-11-06 17:11:20.067850359 [E:onnxruntime:Default, env.cc:254 ThreadMain] pthread_setaffinity_np failed for thread: 4031, index: 9, mask: {9, }, error code: 22 error msg: Invalid argument. Specify the number of threads explicitly so the affinity is not set.
2024-11-06 17:11:20.067833501 [E:onnxruntime:Default, env.cc:254 ThreadMain] pthread_setaffinity_np failed for thread: 4030, index: 8, mask: {8, }, error code: 22 error msg: Invalid argument. Specify the number of threads explicitly so the affinity is not set.
2024-11-06 17:11:20.079936758 [E:onnxruntime:Default, env.cc:254 ThreadMain] pthread_setaffinity_np failed for thread: 4029, index: 7, mask: {7, }, error code: 22 error msg: Invalid argument. Specify the number of threads explicitly so the affinity is not set.
2024-11-06 17:11:20.080029435 [E:onnxruntime:Default, env.cc:254 ThreadMain] pthread_setaffinity_np failed for thread: 4033, index: 11, mask: {11, }, error code: 22 error msg: Invalid argument. Specify the number of threads explicitly so the affinity is not set.
2024-11-06 17:11:20.079941963 [E:onnxruntime:Default, env.cc:254 ThreadMain] pthread_setaffinity_np failed for thread: 4032, index: 10, mask: {10, }, error code: 22 error msg: Invalid argument. Specify the number of threads explicitly so the affinity is not set.
Loading pipeline components...:  29%|██████████████████                                             | 2/7 [00:01<00:02,  1.87it/s]The config attributes {'mid_block_add_attention': True} were passed to AutoencoderKL, but are not expected and will be ignored. Please verify your config.json configuration file.
Loading pipeline components...: 100%|███████████████████████████████████████████████████████████████| 7/7 [00:02<00:00,  2.79it/s]
image2views unet model {'Total': 2567463684, 'Trainable': 0}
/mnt/WujieAITeam/private/wfd/code/Hunyuan3D-1/svrm/ldm/modules/encoders/dinov2/layers/swiglu_ffn.py:43: UserWarning: xFormers is available (SwiGLU)
  warnings.warn("xFormers is available (SwiGLU)")
/mnt/WujieAITeam/private/wfd/code/Hunyuan3D-1/svrm/ldm/modules/encoders/dinov2/layers/attention.py:27: UserWarning: xFormers is available (Attention)
  warnings.warn("xFormers is available (Attention)")
/mnt/WujieAITeam/private/wfd/code/Hunyuan3D-1/svrm/ldm/modules/encoders/dinov2/layers/block.py:35: UserWarning: xFormers is available (Block)
  warnings.warn("xFormers is available (Block)")
None pretrained model for dinov2 encoder ...
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is 768 and using 16 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1024, context_dim is None and using 16 heads.
DEFAULT_RENDERING_KWARGS
{'ray_start': 'auto', 'ray_end': 'auto', 'box_warp': 1.2, 'white_back': True, 'disparity_space_sampling': False, 'clamp_mode': 'softplus', 'sampler_bbox_min': -0.6, 'sampler_bbox_max': 0.6}
SVRMModel has 458.69 M params.
Load model successfully
=====> mv23d model init time: 9.477961540222168
view2mesh model {'Total': 458688965, 'Trainable': 0}
image
stage 2, remove back ground
[HunYuan3D]-[remove background], cost time: 0.0000s
stage 3, image to views
Segmentation fault (core dumped)

**Comment by YUEVII** [comment]
save problem

**Comment by kyrie111** [comment]
> same problem

I’m speechless. It turns out the problem was with the PyTorch version in env_install.sh
reinstall torch==2.0.1 torchvision==0.15.2 torchaudio==2.0.2 works for me

**Comment by YUEVII** [comment]
> > same problem
> 
> I’m speechless. It turns out the problem was with the PyTorch version in env_install.sh reinstall torch==2.0.1 torchvision==0.15.2 torchaudio==2.0.2 works for me

Solved! Thanks a lot!
